Solução do Lab1

Pedropnb opened the pull request:
Funções splitData, mapFunc e reduceFunc implementadas e testadas de acordo com wordcount_test.go.

Pedropnb committed:
Add files via upload

Funções splitData, mapFunc e reduceFunc implementadas e testadas de acordo com wordcount_test.go.

PauloAguiar commented on data.go:
```diff
@@ -0,0 +1,178 @@
+	//cria o slice para leitura com o tamanho desejado
+	data := make([]byte, chunkSize)
+
+	for ;erro != io.EOF;  {//percorre todo o arquivo de entrada
```
Essa notação é equivalente a:

``` go
for erro != io.EOF {
```

PauloAguiar commented on data.go:
```diff
@@ -0,0 +1,178 @@
+								_,_ = arquivo_saida.Write(data_temp)
+								arquivo_saida.Close()
+								//reposiciona o arquivo de leitura para o ponto de corte
+								_,_ = arquivo_ent.Seek(int64(-i+1), 1)
```
Interessante!

PauloAguiar commented on wordcount.go:
```diff
@@ -0,0 +1,124 @@
+	/////////////////////////
+	// Codigo Pedro Nunes Baptista aluno Especial CE-288
+	// essa funcao considera que os registros em input ja estao agrupados por
+	// strings .Key iguais
```
Essa é uma consideração muito forte. Para os testes apresentados no wordcount_test.go ela é válida, mas se executarmos o seu map e reduce no arquivo fornecido ('files/pg1342.txt'), a contagem vai estar incorreta pois não existe nenhuma lógica no map que garanta essa propriedade.

A minha recomendação é utilizar uma estrutura do tipo map[string]int e fazer uma conversão no final.

Pedropnb replied:
Então não entendi corretamente o lab. Entendi que a função Shuffle, que nós não precisaríamos implementar, faria isso antes de distribuir cada conjunto de maps para as funções reduceFunc. Obrigado, vou estudar a solução com a estrutura map.

Pedropnb committed:
Add files via upload

Correção da notação do "for" usado como "while" segundo a sugestão do Paulo.

Pedropnb committed:
Add files via upload

Correção na função reduceFunc para utilizar a estrutura map[string]string e ordenar corretamente as chaves no slice de saída.

PauloAguiar commented:
Pedro,

Com a última alteração o exercício parece estar logicamente correto e a implementação está bem clara. Estou dando merge com o seu branch. Caso você queira fazer alguma alteração, basta submeter uma nova pull request.

Paulo
